object_storage_api/repositories/image.py

```diff
@@ -110,9 +112,18 @@ def update(self, image_id: str, image: ImageIn, session: ClientSession = None) -
         logger.info("Updating image metadata with ID: %s", image_id)
         try:
             image_id = CustomObjectId(image_id)
-            self._images_collection.update_one(
-                {"_id": image_id}, {"$set": image.model_dump(by_alias=True)}, session=session
-            )
+            if update_primary:
+                bulkwrite_update = [
+                    UpdateMany(
+                        filter={"primary": True, "entity_id": image.entity_id}, update={"$set": {"primary": False}}
```

Comment:
How many images can have a value of `True` for the primary field at any one time? My understanding was just one? If that is the case then should we be using `UpdateMany` here if only one needs to be updated or are we going with this for a peace of mind?

Reply:
That's fine then, thanks for clarifying.